types/define-properties/index.d.ts

```diff
@@ -0,0 +1,27 @@
+// Type definitions for define-properties 1.1
+// Project: https://github.com/ljharb/define-properties#readme
+// Definitions by: ExE Boss <https://github.com/ExE-Boss>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+declare namespace defineProperties {
+	/**
+	 * Whether the current environment correctly supports property descriptors.
+	 */
+	const supportsDescriptors: boolean;
+}
+
+/**
+ * Defines new properties in `map` as non-enumerable if they don't already
+ * exist on `object`.
+ *
+ * @param object The object to define non-enumerable properties on.
+ * @param map The map of newly defined properties.
+ * @param predicates The optional predicates map, return `true` to override existing properties on `object`.
+ */
+declare function defineProperties<K extends keyof any>(
```

Comment:
why is this any? The K must be a Property Key, ie a string/number/symbol/bigint

Reply:
It extends `keyof any`, which resolves to `string | number | symbol`.

Also, last I checked, `bigint` couldn’t be used to index properties directly in TypeScript (unless they changed that).

Reply:
true, that hasn’t been fixed yet.